### ISSUE TITLE
Mint 20.3 Cinnamon Does not Reconnect on Ethernet After Sleep

### ISSUE THREAD

**Issue by maciekskontakt** [linuxmint/cinnamon#12052]
### Distribution

Mint 20.3

### Package version

5.2.7

### Graphics hardware in use

Asus PN-50 Barebone PC Hardware (Not GraphicsBut Ethernet)

### Frequency

Always

### Bug description

Ethernet built-in. After sleep it does not reconnect when Linux is woken up. No manual forcing helps, occasionally reconnecting cable may help but not always. Only system restart helps reliably. This does not occur when using USB Ethernet dongle. So it looks like driver for builtin hardware in ASUS barebone does not behave correctly or it does not use proper sequence of operations to re-connect to router. The same router works with newer Mint version even on old legacy hardware (MacBook Air from 2011) via USB dongle with no issues. The router works with about 20 devices and 6 computers on different OS's so, this should not be an issue related to router, but this particular Mint version with it's updates on Asus hardware.

It worked perfectly when installed in 2022 and it does not now. Linux Mint 20.3 (5.2.7) Cinnamon after many kernel updates by software update through update manager. Kernel is 5.4.0-170-generic.

### Steps to reproduce

Get the hardware and get it to sleep and then try to wake up and see if Ethernet connection is restored (it is not)

### Expected behavior

The same as with external dongles. After Ethernet wake up reconnecting should work every time.

**Comment by P-Chris75** [comment]
I have the same error under Mate, on my Acer Extensa 5635zg laptop model. However, in the case of my more modern DELL 5490 laptop, it no longer exists. Acer still has an old P9700 processor, Dell already has an 8th generation i5 processor.

**Comment by SjoerdV** [comment]
@maciekskontakt you are not connected to a 'hidden' SSID by any chance, because then autoconnect would not work.

**Comment by maciekskontakt** [comment]
What SSID? I clearly specified in description it is not WiFi as I do not use WiFi on this PC. It is Ethernet.

**Comment by maciekskontakt** [comment]
|  What SSID? I clearly specified in description it is not WiFi as I do not use WiFi on this PC. It is Ethernet. WiFi is blocked on my Mint computers. I am not sure why everybody assumes people use WiFi only. I run high security network as work as It archotect in NYC bank and run also business. All of that MUST be on Ethernet that cannot be spoofed easily or blocked by radio means like WiFi.
  |



    On Tuesday, February 27, 2024 at 06:48:47 PM EST, SjoerdV ***@***.***> wrote:  
 
 


@maciekskontakt you are not connected to a 'hidden' SSID by any chance, because then autoconnect would not work.

**Comment by maciekskontakt** [comment]
And for the record this is content of script i have to run every time after waking PC up so it connects properly:

sudo ethtool enp2s0f0
sudo lshw -C network
inxi -Nn 
sudo modprobe -r r8168 && sudo modprobe r8168 && sudo systemctl restart network-manager.service 




    On Tuesday, February 27, 2024 at 07:59:55 PM EST, ***@***.*** ***@***.***> wrote:  
 
  
|  What SSID? I clearly specified in description it is not WiFi as I do not use WiFi on this PC. It is Ethernet. WiFi is blocked on my Mint computers. I am not sure why everybody assumes people use WiFi only. I run high security network as work as It archotect in NYC bank and run also business. All of that MUST be on Ethernet that cannot be spoofed easily or blocked by radio means like WiFi.
  |



    On Tuesday, February 27, 2024 at 06:48:47 PM EST, SjoerdV ***@***.***> wrote:  
 
 


@maciekskontakt you are not connected to a 'hidden' SSID by any chance, because then autoconnect would not work.

**Comment by claudiux** [comment]
This bug is not specific to Mint, but there is a way to solve it.

You need to know the driver of your Ethernet card: `inxi -Fxz` then look at the Network section. On my computer:

```
Network:
  Device-1: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet
    vendor: Gigabyte driver: r8169 v: kernel port: e000 bus-ID: 02:00.0
  IF: enp2s0 state: up speed: 1000 Mbps duplex: full mac: <filter>
  IF-ID-1: virbr0 state: down mac: <filter>
```
Driver is **r8169** here.

Now create the script `/lib/systemd/system-sleep/ethernet-wakeup` with the root rights, which contains:

```
#!/bin/sh

# Wake up Ethernet !

case $1/$2 in
	pre/*)
		modprobe -r r8169
		sleep 1
		;;
	post/*)
		modprobe r8169
		;;
esac
```
(Replace **r8169** by your own driver)

Make this script executable: `sudo chmod +x /lib/systemd/system-sleep/ethernet-wakeup`

Reboot the computer. After the next standby, Ethernet restarts automatically!

**Comment by maciekskontakt** [comment]
I understand it may not be specific to Mint, but I use Mint distribution and I have no ability or time to check other distributions. Hence i report this against Mint to enable to report to those responsible for kernel or driver fix as I did not provide my own driver.
Is there a way to have that script included in distribution and properly scripted command that extracts driver putting into parametrized script you suggested? User is least interested in becoming Linux geek (although I have scripted in Bash and other UNIX shell scripts many times for commercial purposes on servers in the past). It would be nice to have that workaround included for desktop users. Extracting parameter for modprobe from line should not be a problem:
vendor: ASUSTeK driver: r8168 v: 8.048.00-NAPI port: fc00 bus ID: 02:00.0 

The output for driver is:
Network:
  Device-1: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet 
  vendor: ASUSTeK driver: r8168 v: 8.048.00-NAPI port: fc00 bus ID: 02:00.0 
  IF: enp2s0f0 state: up speed: 1000 Mbps duplex: full mac: <filter> 
  Device-2: Intel Wi-Fi 6 AX200 driver: iwlwifi v: kernel port: f000 
  bus ID: 03:00.0 
  IF: wlp3s0 state: down mac: <filter> 

Thanks,-Maciek


    On Wednesday, February 28, 2024 at 04:33:29 AM EST, claudiux ***@***.***> wrote:  
 
 


This bug is not specific to Mint, but there is a way to solve it.

You need to know the driver of your Ethernet card: inxi -Fxz then look at the Network section. On my computer:
Network:
  Device-1: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet
    vendor: Gigabyte driver: r8169 v: kernel port: e000 bus-ID: 02:00.0
  IF: enp2s0 state: up speed: 1000 Mbps duplex: full mac: <filter>
  IF-ID-1: virbr0 state: down mac: <filter>

Driver is r8169 here.

Now create the script /lib/systemd/system-sleep/ethernet-wakeup with the root rights, which contains:
#!/bin/sh

# Wake up Ethernet !

case $1/$2 in
	pre/*)
		modprobe -r r8169
		sleep 1
		;;
	post/*)
		modprobe r8169
		;;
esac

(Replace r8169 by your own driver)

Make this script executable: sudo chmod +x /lib/systemd/system-sleep/ethernet-wakeup

Reboot the computer. After the next standby, Ethernet restarts automatically!

**Comment by claudiux** [comment]
Your kernel is a 5.4.
Your Ethernet card is not well supported before kernel 5.8. (Yes, this is not a Mint-specific bug.)

You can try one of these solutions:

1rst solution: Upgrade your distro to a most recent version.
2nd solution: On the same distro, use a 5.15 version of Linux kernel (open mintupdate, in menu View select Linux kernels).
3rd solution: Install the r8168-dkms package.

**Comment by maciekskontakt** [comment]
I understand, but here is the deal: I am on LTS version and I do not plan to upgrade few months before it expires to next LTS version. I do not experiment with non-LTS versions or do updates frequently (except required from Software Center). I assume LTS means something. I use Mint for serious work as alternative as I moved away from macOS (after 15 years on Mac OSX/macOS) and Windows (in around 2005). I used Linux SuSE an Caldera before that time, but my time for supporting Linux drivers (I used to support some modem drivers) is long gone.
Thanks

    On Wednesday, February 28, 2024 at 08:33:21 PM EST, claudiux ***@***.***> wrote:  
 
 


Your kernel is a 5.4.
Your Ethernet card is not well supported before kernel 5.8. (Yes, this is not a Mint-specific bug.)

You can try one of these solutions:

1rst solution: Upgrade your distro to a most recent version.
2nd solution: On the same distro, use a 5.15 version of Linux kernel (open mintupdate, in menu View select Linux kernels).
3rd solution: Install the r8168-dkms package.

**Comment by claudiux** [comment]
Version 5.15 of kernel is in your LTS. After that, the choice is yours...

**Comment by maciekskontakt** [comment]
When I downloaded Mint I paid attention to version of Linux Mint LTS. I am on LTS. What kernel it is I do not care. Do you suggest that Mint is LTS from 2021, but kernel is not? That would sound odd. Plese explain how it works. I do not download kernels except mandatory minor version updates. I do download system that is supposed to be LTS. I do not work on kernel - I do eork on system. Mint 20.3 I believe. What’s behind it is not user business.
    On Thursday, February 29, 2024, 06:50:32 AM EST, claudiux ***@***.***> wrote:  
 
 


Version 5.15 of kernel is in your LTS. After that, the choice is yours...

**Comment by maciekskontakt** [comment]
20.3 Una Cinnamon Ubuntu Focal Long term support release (LTS), supported until April 2025.
Please, chek your own website and do not suggest user should do anything with kernel which is part of OS:
https://www.linuxmint.com/download_all.php
    On Thursday, February 29, 2024, 12:39:09 PM EST, ***@***.*** ***@***.***> wrote:  
 
  When I downloaded Mint I paid attention to version of Linux Mint LTS. I am on LTS. What kernel it is I do not care. Do you suggest that Mint is LTS from 2021, but kernel is not? That would sound odd. Plese explain how it works. I do not download kernels except mandatory minor version updates. I do download system that is supposed to be LTS. I do not work on kernel - I do eork on system. Mint 20.3 I believe. What’s behind it is not user business.
    On Thursday, February 29, 2024, 06:50:32 AM EST, claudiux ***@***.***> wrote:  
 
 


Version 5.15 of kernel is in your LTS. After that, the choice is yours...
